Switch to using Eigen for Quaternion and Matrix.

clalancette opened the pull request:
This should fix #6, though to be conservative we may want to merge this onto a melodic-devel branch, rather than kinetic-devel.

Signed-off-by: Chris Lalancette <clalancette@osrfoundation.org>

clalancette committed:
Switch to using Eigen for Quaternion and Matrix.

Signed-off-by: Chris Lalancette <clalancette@osrfoundation.org>

sloretz commented on collada_urdf/CMakeLists.txt:
```diff
@@ -48,13 +48,33 @@ endif()
+# We don't build anything here, but in order to export the right
+# build flags in catkin_package(), we still have to find Eigen3 here.  Note
+# that this is somewhat complicated because of our need to support Ubuntu
+# all the way back to saucy.  First we look for the Eigen3 cmake module
```
[REP 3](http://www.ros.org/reps/rep-0003.html#kinetic-kame-may-2016-may-2021) says kinetic targets xenial and above. [libeigen3-dev](https://packages.ubuntu.com/search?keywords=libeigen3-dev) is in trusty and above. I think this block can be replaced with `find_package(Eigen3 REQUIRED)`

clalancette replied:
Ah, good point.  It looks like Kinetic used to support Wily, but we've long stopped worrying about that.  Fixed in 4edadbf

sloretz commented on collada_urdf/CMakeLists.txt:
```diff
@@ -48,13 +48,33 @@ endif()
+  set(EIGEN3_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS})
+endif()
+
+# Note that eigen 3.2 (on Ubuntu Wily) only provides EIGEN3_INCLUDE_DIR,
```
Same here, xenial through bionic is `3.3.*`

sloretz reviewed (commented):
Sorry it took me so long to review this. Looks good after a couple CMake changes.

sloretz commented:
I notice #6 has a melodic label. I can't think of anything that would block targeting this at kinetic. Can you think of anything?

clalancette commented:
> I notice #6 has a melodic label. I can't think of anything that would block targeting this at kinetic. Can you think of anything?

I agree with you; this is all internal code, so we should be able to merge this for kinetic as well.  There is a slight chance for downstream breakage if downstream packages were depending on this one to pull in TF, but I think that is acceptable (and should be relatively easy to fix).

clalancette committed:
Simplify Eigen3 dependency.

We only support Xenial+ now, so we don't need to worry about
backwards compatibility anymore.

Signed-off-by: Chris Lalancette <clalancette@osrfoundation.org>

clalancette commented:
Great, thanks!  Merging.
